specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,428 @@
+* **`GetForUser()`** -- Synchronous equivalent to GetForUserAsync().
+* **`GetForUnpackaged()`** -- Equivalent functionality for unpackaged apps. See
+  [3.5. Unpackaged app data stores](#35-unpackaged-app-data-stores).
+
+## 3.2. *Path properties
+
+New *Path equivalents to Windows.Storage.ApplicationData's *Folder but returning a `String` rather than `StorageFolder`.
+
+## 3.3. Reliable access
+
+`Windows.Storage.ApplicationData.Current` only works for packaged apps **running in an AppContainer**. Processes not running in an AppContainer (e.g. MediumIL) must use the equivalent `Windows.Management.Core.ApplicationDataManager.CreateForPackageFamily(Windows.ApplicationModel.Package.Current.Id.FamilyName)`.
+
+Microsoft.Windows.Storage.ApplicationData provides consistent behavior as developers expect e.g. `ApplicationData.Current` works regardless if the caller's running in an AppContainer or not.
+
+## 3.4. Per-Machine data store
+
+Package families can optionally have a per-machine data store.
+
+### 3.4.1. Machine Path/Folder
+
+The path for a package family’s "Machine Folder" is located at `%ProgramData%\Microsoft\Windows\AppRepository\ApplicationData\...packagefamilyname...\Machine`.
+
+This directory is ACL’d similarly to a package’s System Metadata directory (`%ProgramData%\Microsoft\Windows\AppRepository\Packages\...pkgfullname...`) e.g.
+
+```
+BUILTIN\Users:(OI)(CI)(Rc,S,RD,REA,X,RA)
+S-1-15-3-2977037414-864741429-1129033548-1928484290-1803339615-1058153653-556172075:(OI)(CI)(RX)
+BUILTIN\Users:(OI)(CI)(R)
+NT SERVICE\TrustedInstaller:(I)(F)
+NT SERVICE\TrustedInstaller:(I)(CI)(IO)(F)
+NT AUTHORITY\SYSTEM:(I)(F)
+NT AUTHORITY\SYSTEM:(I)(OI)(CI)(IO)(F)
+BUILTIN\Administrators:(I)(RX)
+BUILTIN\Administrators:(I)(OI)(CI)(IO)(GR,GE)
+S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204:(I)(RX)
+S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204:(I)(OI)(CI)(IO)(GR,GE)
+```
+
+### 3.4.2. Manifested Opt-In
+
+The "Machine Folder" is only available for packages that explicitly opt-in to enabling it via appxmanifest.xml
+
+```xml
+<Package>
+  ...
+  <Properties>
+    <appdata:ApplicationData>
+      <MachineFolder/>
+    </appdata:ApplicationData>
+    ...
+```
+
+### 3.4.3. Machine Path Creation/Deletion
+
+A new Undocked Deployment Extension Handler (DEH) creates the Machine path when a package in the package family is installed, if it doesn't already exist.
+
+The Machine path is removed when the last package in the package family is removed from the machine.
+
+### 3.4.4. APIs
+
+New properties and methods to access per-machine data:
+
+* ApplicationDataLocality.Machine
+* ClearMachineFolderAsync()
+* IsMachinePathSupported
+* MachineFolder
+* MachinePath
+
+**NOTE:** The per-machine APIs require the caller has the package family registered for the user -or- is running as SYSTEM.
```

Comment:
When would this result in access denied? Like, "An app fooing the bar with zot does not have access."

Reply:
You're right. I misread the ACL not realizing Administrator and User has access (regardless packaged or not).

Access should be governed by the path's ACL. The API shouldn't make presumptions (especially not false ones). After the path's just a string, not until _access_ managed per ACLs is Access/Denied relevant.

RECOMMEND: Remove line 148.